libqtile/layout/xmonad.py

```diff
@@ -176,32 +176,21 @@ class MonadTall(_SimpleLayoutBase):
             " bottom - at the bottom of the stack,"),
     ]
 
-    def __init__(self, new_at_current=None, **config):
+    def __init__(self, **config):
         _SimpleLayoutBase.__init__(self, **config)
-        # is key new_client_position set by user in config file ?
-        try:
-            user_set_ncp = self.new_client_position
-        except AttributeError:
-            user_set_ncp = None
         self.add_defaults(MonadTall.defaults)
         if self.single_border_width is None:
             self.single_border_width = self.border_width
         if self.single_margin is None:
             self.single_margin = self.margin
         self.relative_sizes = []
         self.screen_rect = None
-        if new_at_current is not None:
-            logger.warning('MonadTall layout key new_at_current is deprecated.')
-            # manage priority if both key are set,
-            if user_set_ncp is None:
-                logger.warning('Please use new_client_position instead.')
-                # set new_client_position considering new_at_current value.
-                if new_at_current:
-                    self.new_client_position = "before_current"
-                else:
-                    self.new_client_position = "after_current"
-            else:  # new_at_current value will not be considered.
-                logger.warning('Please use only new_client_position.')
+        try:
+            if hasattr(self, 'new_at_current'):
+                logger.warning('Monad<x> layout key new_at_current is deprecated.'
+                               ' Please use new_client_position instead.')
+        except AttributeError:
```

Comment:
You don't need the `except` bit here, hasattr() doesn't throw if an attribute isn't present, it just returns false. That said, you might be able to drop this whole bit and just rely on the `migrate` code to handle this for people. Hopefully we'll have a better story about automatic migration for the next release.